openapi-cli/src/main/java/io/ballerina/openapi/generators/service/ListenerGenerator.java

```diff
@@ -0,0 +1,168 @@
+import io.swagger.v3.oas.models.servers.ServerVariables;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.List;
+
+import static io.ballerina.openapi.generators.GeneratorConstants.CONFIG;
+import static io.ballerina.openapi.generators.GeneratorConstants.HOST;
+import static io.ballerina.openapi.generators.GeneratorConstants.NEW;
+
+/**
+ * This util class for processing the mapping in between openAPI server section with ballerina listeners.
+ *
+ * @since 2.0.0
+ */
+public class ListenerGenerator {
+    private static final int httpPort = 80;
+    private static final int httpsPort = 443;
+    private String basePath = "/";
+
+    public ListenerGenerator() {
+    }
+
+    public String getBasePath() {
+        return basePath;
+    }
+
+    /**
+     * Generate listener accoring to the given server details.
+     *
+     * @param servers  OAS server details
+     * @return         {@link ListenerDeclarationNode} for server.
+     * @throws BallerinaOpenApiException when process break with exception
+     */
+    public ListenerDeclarationNode getListenerDeclarationNodes(List<Server> servers) throws BallerinaOpenApiException {
+        // Assign host port value to listeners
+        String host;
+        int port;
+        Server server = servers.get(0);
+        if (server.getVariables() != null) {
+            ServerVariables variables = server.getVariables();
+            URL url;
+            try {
+                String resolvedUrl = GeneratorUtils.buildUrl(server.getUrl(), variables);
+                url = new URL(resolvedUrl);
+                host = url.getHost();
+                basePath = url.getPath();
+                port = url.getPort();
+                boolean isHttps = "https".equalsIgnoreCase(url.getProtocol());
+
+                if (isHttps) {
+                    port = port == -1 ? httpsPort : port;
+                } else {
+                    port = port == -1 ? httpPort : port;
+                }
+            } catch (MalformedURLException e) {
+                throw new BallerinaOpenApiException("Failed to read endpoint details of the server: " +
+                        server.getUrl(), e);
+            }
+        } else if (!server.getUrl().isBlank()) {
+            String[] split = server.getUrl().trim().split("/");
+            basePath =  "/" + split[split.length - 1];
+            host = "localhost";
+            port = 9090;
+        } else {
+            host = "localhost";
+            port = 9090;
+        }
+        return getListenerDeclarationNode(port, host, " ep0");
+    }
+
+    public static ListenerDeclarationNode getListenerDeclarationNode(Integer port, String host, String ep) {
+
+        // Take first server to Map
+        Token listenerKeyword = AbstractNodeFactory.createIdentifierToken("listener");
+        // Create type descriptor
+        Token modulePrefix = AbstractNodeFactory.createIdentifierToken(" http");
```

Comment:
If you need to consider leading/trailing whitespaces when creating tokens, you can use `AbstractNodeFactory.createIdentifierToken(String text, MinutiaeList leadingMinutiae, MinutiaeList trailingMinutiae);`. Revisit all other similar places to avoid defining constants with whitespaces.